www/js/diary/current.js

```diff
@@ -82,31 +82,13 @@
     return hours + ':' + minutes + ' ' + amOrPm;
   };
 
-  var deg2rad = function(deg) {
-    return deg * (Math.PI/180);
-  };
-
-  // Haversine formula
-  var getMeterfromLglat = function(curr_lglat, last_lglat) {
-    lat1 = curr_lglat[1];
-    lg1 = curr_lglat[0];
-    lat2 = last_lglat[1];
-    lg2 = last_lglat[0];
-    var R = 6371e3; // earth radius in metres
-    var dLat = deg2rad(lat2-lat1); 
-    var dLg = deg2rad(lg2-lg1); 
-    var a = Math.sin(dLat/2) * Math.sin(dLat/2) +
-            Math.cos(deg2rad(lat1)) * Math.cos(deg2rad(lat2)) * 
-            Math.sin(dLg/2) * Math.sin(dLg/2); 
-    var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a)); 
-    var d = R * c;
-    return d;
-  };
-
   var getSpeed = function(curr_lglat, last_lglat, curr_ts, last_ts) {
-    distance = getMeterfromLglat(curr_lglat, last_lglat);
+    var curr_latlng = L.latLng(curr_lglat[1],curr_lglat[0]);
+    var last_latlng = L.latLng(last_lglat[1],last_lglat[0]);
+    var meters = curr_latlng.distanceTo(last_latlng);
+    console.log("Distance To", meters);
```

Comment:
nit/fyi: using the Logger service will log to the database log that can be emailed in case we find bugs. Might want to do that at least for important logs.